internal/app/machined/pkg/controllers/runtime/security_state.go

```diff
@@ -0,0 +1,152 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package runtime
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/foxboron/go-uefi/efi"
+	"github.com/foxboron/go-uefi/efi/attributes"
+	"github.com/foxboron/go-uefi/efi/pecoff"
+	"github.com/foxboron/go-uefi/efi/util"
+	"go.mozilla.org/pkcs7"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/mount"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	runtimeres "github.com/siderolabs/talos/pkg/machinery/resources/runtime"
+)
+
+// SecurityStateController is a controller that updates the security state of Talos.
+type SecurityStateController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *SecurityStateController) Name() string {
+	return "runtime.SecurityStateController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *SecurityStateController) Inputs() []controller.Input {
+	return nil
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *SecurityStateController) Outputs() []controller.Output {
+	return []controller.Output{
+		{
+			Type: runtimeres.SecurityStateType,
+			Kind: controller.OutputExclusive,
+		},
+	}
+}
+
+// Run implements controller.Controller interface.
+// nolint:gocyclo
+func (ctrl *SecurityStateController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
+	select {
+	case <-ctx.Done():
+		return nil
+	case <-r.EventCh():
+	}
+
+	spec := runtimeres.NewSecurityStateSpec(runtimeres.NamespaceName)
+
+	if efi.GetSecureBoot() && !efi.GetSetupMode() {
+		spec.TypedSpec().SecureBooted = true
+	}
+
+	_, data, err := attributes.ReadEfivarsWithGuid(sdboot.LoaderEntryDefaultName, *util.StringToGUID(sdboot.SystemdBootGUIDString))
+	if err != nil {
+		logger.Info("Failed to read systemd-boot loader entry", zap.Error(err))
+	}
+
+	loaderDefaultEntryName := data.String()
+
+	// lets mount the EFI partition and read the loader entry
+	if opErr := mount.PartitionOp(ctx, "", constants.EFIPartitionLabel, func() error {
+		if _, statErr := os.Stat(filepath.Join(constants.EFIMountPoint, "EFI", "Linux", loaderDefaultEntryName)); statErr == nil {
```

Comment:
I feel this is not really reliable, given the boot from PXE/ISO. we need some way to get that cert without going from the bootloader efi vars

Reply:
an easy fix is to put it into rootfs at a known path and we read from it (we could read the efi db variable, but that could contain not just the UKI signing keys, I couldn't think of a way to distinguish between them) :sweat_smile:

Reply:
if we can't do that right now, let's skip it

Reply:
what I'm trying to say is probably from an attestation point of view, I would rather see the keys which are authorized for secure boot, and the key which was used this time (if possible), but not the key the UKI is signed with... as we can be mislead to read the wrong UKI

Reply:
> and the key which was used this time (if possible)

not sure i follow

> but not the key the UKI is signed with... as we can be mislead to read the wrong UKI
there's only a single UKI booted and that the boot entry so this was just parsing the cert from the PE signature

I think i can add another field that lists all keys from UEFI db though (authorized keys for secureboot)

Reply:
I mean there was a secure boot process which verified the UKI, sd-boot, etc. there were some keys which were used to verify the signatures, which key it were?

> there's only a single UKI booted and that the boot entry so this was just parsing the cert from the PE signature

I don't think this holds true for ISO/PXE boot. For PXE boot, there's no UKI to read at all. 

So as long as we can't provide a reliable solution, we can skip it for now (and add it later)

Reply:
> I mean there was a secure boot process which verified the UKI, sd-boot, etc. there were some keys which were used to verify the signatures, which key it were?

that's the key used to sign the UKI, the public part is loaded into UEFI db, and firmware just verfies it's signed by the key in db

Reply:
> I don't think this holds true for ISO/PXE boot. For PXE boot, there's no UKI to read at all.

correct, but once talos is installed the data would get populated

Reply:
that's what I'm after - if we don't have a solution right now, let's keep it simple and skip this key. SecureBoot status is all we need right now